### PR TITLE
Odin JS Weather API Project

### PR DESCRIPTION
**Pull request by kwizl**
# Consuming API
- This project is a weather forecast site using the OpenWeatherAPI. The user should be able to search for a specific location and displaying the details of the weather. The appication does this by consuming the API and showing the desired data to the user.

### CODE REVIEW

File: dist/css/style.css [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  padding-top: 40px;
+}
+
+.error {
+  color: white;
+  font-size: 20px;
+  text-align: center;
+}
+
+#content {
```
Comment (uryelah) [inline dist/css/style.css]: - [x] Remember CSS good practices, avoid styling elements using ids, prefer classes.

File: src/js/models/search.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+          message: 'Invalid data',
+        };
+      } else {
+        const response = await fetch(`https://api.openweathermap.org/data/2.5/weather?q=${this.query}&&units=metric&appid=5b2f48f1d0bbdc18d1a72fba0922419e`, { mode: 'cors' });
+        const result = await response.json();
+        display.apiData(result);
+        display.errorErase();
+      }
+    } catch (error) {
+      display.errorDisplay(error.message);
```
Comment (uryelah) [inline src/js/models/search.js]: - [x] Make sure to display a friendly error to the user, set a readable message for when the user tries to search for an inexistent city, at the moment it shows 'e.main is undefined' and 'City: undefined'.

File: src/js/views/display.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    content.appendChild(divResults);
+    divResults.appendChild(h2);
+    divResults.appendChild(p1);
+    divResults.appendChild(p2);
+    divResults.appendChild(p3);
+    divResults.appendChild(p4);
+  };
+
+  const apiData = (data) => {
```
Comment (uryelah) [inline src/js/views/display.js:29]: - [x] Let's also change something visual within the app according to the received data. 
It's a requirement to change the appearance of the app according to the data received, like the background reflecting the weather. 
appearance_requirement

**Review by uryelah — CHANGES_REQUESTED** [review]
Hi **Martin Njoroge**, I'm Sarah your reviewer for this PR 👩‍🚀 

Good job! Your code is beautifully organized and you've added a very charming button interaction.

Now, you are still missing a couple of core requirements for this project, namely the app look changing according to the received data and the toggle button between Celcius and Fahrenheit.

Check the pointers below and keep on coding! 🏃 💨 

# Required Changes 🏗️ 🐛 💡

- [x] Add a toggle control to determine if the result will be shown in Celsius or Fahrenheit.
search_requirement

- [x] Change the app appearance according to the fetched data.

- [ ] [OPTIONAL] Add a loader to show the user while the data is being fetched.

- [ ] [OPTIONAL] If you want a challenge: fetch the current geolocation of the user when the app is first accessed and use that location as the first search when the app is loaded.

-----

 **Kindly fix the pointed out issues and comments below before requesting the next code review**

------------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Review by rOluochKe — APPROVED** [review]
Hello @kwizl 
### Good job :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next one.
- Happy hacking!

### Optional
- You can write more scripts to automate some of the tasks for you like, see the scripts below
```
"scripts": {
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
    "start": "webpack-dev-server --mode development --open" // optional
  },
```

File: src/js/views/display.js (hunk 3 above)
Comment (kwizl) [reply src/js/views/display.js:29]: Implemented all corrections except the two optional ones. Thanks for the detailed feedback

**Pull request merged**